gpkit/constraints/sgp.py

```diff
@@ -246,3 +274,24 @@ def gp(self, x0=None, mutategp=False):
             gp = GeometricProgram(self.cost, gp_constrs, self.substitutions)
             gp.x0 = x0  # NOTE: SIDE EFFECTS
         return gp
+
+    def penalty_ccp(self, exp=10.):
+        "Returns the penalty convex-concave form of this SP."
+        signomials = []
+        gp_constrs = []
+        for constr in self.flat(constraintsets=False):
+            if isinstance(constr, (SingleSignomialEquality,
+                                   SignomialInequality)):
+                signomials.append(constr)
+            else:
+                gp_constrs.append(constr)
+        slack = VectorVariable(len(signomials), 's')
```

Comment:
You should do this in a `with NamedVariables(foo):` environment to ensure the variable name is unique. Maybe `foo = CCP`?